src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/Json/XmlObjectSerializerWriteContextComplexJson.cs

```diff
@@ -326,6 +326,12 @@ private void VerifyType(DataContract dataContract, Type declaredType)
                 scopedKnownTypes.Pop();
             }
         }
+
+        internal static void WriteJsonNameWithMapping(XmlWriterDelegator xmlWriter, XmlDictionaryString[] memberNames, int index)
```

Comment:
Nevermind @stephentoub already pointed me to who calls it. I do agree with trying to add some validation where we can.